### PR TITLE
Only consider files ending with .sol and not starting with ~ in synta…

### LINKED ISSUE #3742: Syntax test should skip vim swap files and potentially others

### PR DESCRIPTION
**Pull request by ekpyron**
…x tests.

Fixes #3742.

Are there any other naming schemes for temporary files that will still result in temporary files being considered valid tests?

### CODE REVIEW

File: test/libsolidity/SyntaxTest.cpp [hunk 1]
```diff
@@ -220,7 +220,12 @@ int SyntaxTest::registerTests(
 			fs::directory_iterator(fullpath),
 			fs::directory_iterator()
 		))
-			numTestsAdded += registerTests(*sub_suite, _basepath, _path / entry.path().filename());
+			if (
```
Comment (chriseth) [inline test/libsolidity/SyntaxTest.cpp]: Please format as
```
if (fs::is_directory(entry.path()) || (
    ... &&
))
``

File: test/libsolidity/SyntaxTest.cpp [hunk 2]
```diff
@@ -220,7 +220,12 @@ int SyntaxTest::registerTests(
...
+				fs::is_directory(entry.path()) ||
+				(entry.path().extension().string() == ".sol" &&
+				!boost::starts_with(entry.path().filename().string(), "~"))
```
Comment (chriseth) [inline test/libsolidity/SyntaxTest.cpp]: Hm, I thought about excluding files that start with `.`.

**Review by chriseth — CHANGES_REQUESTED** [review]
Please reduce code duplication.

**Comment by ekpyron** [comment]
@chriseth Requested changes done: moved filename filter to a static function in SyntaxTest; also filter files starting with ".". Code formatting request becomes obsolete.

**Review by chriseth — APPROVED** [review]

**Pull request merged**